dracut/30ignition/ignition-complete.target

```diff
@@ -7,6 +7,7 @@
 Description=Ignition Complete
 OnFailure=emergency.target
 OnFailureJobMode=isolate
+Before=initrd.target
```

Comment:
I've tested failures in both the files and disks stage, not the mount stage. Though I should test at least the umount stage explicitly since it's an `ExecStop`, not `ExecStart` like the other ones.

Reply:
Ahh good catch. It's indeed not erroring on the `ExecStop`.

ISTM like at least for services that use `ExecStop`, we need to keep the `OnFailure=emergency.target` bit (though I'd rather we keep it in all of them for the same reason as https://github.com/coreos/ignition-dracut/pull/72#discussion_r271890913). When we isolate to the new target, `initrd.target` and `ignition-complete.target` are stopped independently of the services that were started by them in the first place (e.g. `ignition-mount.service`), so a failure in those stop jobs won't trigger the `OnFailure=` in `ignition-complete.target`. I just tried this locally:

```
[root@lux ~]# cat > /run/systemd/system/foo.target
[Unit]
OnFailure=bar.service
[root@lux ~]# cat > /run/systemd/system/bar.service
[Service]
ExecStart=/bin/true
[root@lux ~]# cat > /run/systemd/system/pre-foo.service
[Unit]
Before=foo.target
[Service]
ExecStart=/bin/true
ExecStop=/bin/false
[root@lux ~]# mkdir /run/systemd/system/foo.target.requires
[root@lux ~]# ln -sf ../pre-foo.service /run/systemd/system/foo.target.requires/pre-foo.service
[root@lux ~]# systemctl daemon-reload
[root@lux ~]# systemctl start foo.target
[root@lux ~]# systemctl stop foo.target
```

What happened when I did `systemctl start foo.target` is that it started `pre-foo.service`, but `systemctl stop foo.target` doesn't automatically stop `pre-foo.service`, so it doesn't fail and thus doesn't run its `OnFailure=`. OTOH, adding `OnFailure=bar.service` to `pre-foo.service` and doing `systemctl stop pre-foo.service` does trigger `bar.service`.

Now, looking at https://github.com/coreos/bootengine/blob/107ce54747f739dad4b56f9753852341bb759a4d/dracut/30ignition/sysroot-boot.service, I noticed it doesn't have an explicit `OnFailure=`. Is it still the case that on CL if the `ExecStop=` fails, it fails the whole boot?

Reply:
Not sure, given the weirdness and subtlety around `ExecStop` maybe it's worth breaking umount into its own unit and just ordering it with the rest of the units?